### PR TITLE
[TF] Add scratch implementation of batch norm

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_convolutional-modern/batch-norm.md [hunk 1]
```diff
@@ -434,13 +442,13 @@ class BatchNorm(tf.keras.layers.Layer):
...
-            mean=mean,
-            variance=variance,
-            offset=self.beta,
-            scale=self.gamma,
-            variance_epsilon=1e-5)
+            moving_mean=mean,
+            moving_var=variance,
+            beta=self.beta,
+            gamma=self.gamma,
+            eps=1e-5)
```
Comment (astonzhang) [inline chapter_convolutional-modern/batch-norm.md:451]: Generally we want to keep coding block short so it saves space. Can you be greedy with line breaks when listing arguments?

Comment (astonzhang) [reply chapter_convolutional-modern/batch-norm.md:451]: It also applied to other places.

**Comment by mli** [comment]
Job d2l-en/PR-1074/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1074/

**Comment by mli** [comment]
Job d2l-en/PR-1074/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1074/

File: chapter_convolutional-modern/batch-norm.md (hunk 1 above)
Comment (terrytangyuan) [reply chapter_convolutional-modern/batch-norm.md]: Thanks. Fixed.

**Comment by mli** [comment]
Job d2l-en/PR-1074/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1074/

**Comment by astonzhang** [comment]
merging this @mli
